src/Nethermind/Nethermind.JsonRpc/JsonRpcService.cs

```diff
@@ -333,12 +333,12 @@ private JsonRpcErrorResponse GetErrorResponse(string? methodName, int errorCode,
 
             methodName = methodName.Trim();
 
-            ModuleResolution result = _rpcModuleProvider.Check(methodName, rpcEndpoint);
+            ModuleResolution result = _rpcModuleProvider.Check(methodName, context);
             return result switch
             {
                 ModuleResolution.Unknown => ((int?) ErrorCodes.MethodNotFound, $"Method {methodName} is not supported"),
-                ModuleResolution.Disabled => (ErrorCodes.InvalidRequest, $"{methodName} found but the containing module is disabled, consider adding module to JsonRpcConfig.EnabledModules"),
-                ModuleResolution.EndpointDisabled => (ErrorCodes.InvalidRequest, $"{methodName} found but disabled for {rpcEndpoint}"),
+                ModuleResolution.Disabled => (ErrorCodes.InvalidRequest, $"{methodName} found but the containing module is disabled for the url, consider adding module for additional url in JsonRpcConfig.AdditionalRpcUrls or to JsonRpcConfig.EnabledModules"),
+                ModuleResolution.EndpointDisabled => (ErrorCodes.InvalidRequest, $"{methodName} found but disabled for {context.RpcEndpoint}"),
```

Comment:
Cool, one thing still to add into the message now would be the url (or maybe just port?) for which it is disabled, so user can see that maybe he is querying wrong one.

Reply:
Yeah for sure. I will do that.